Automattic/pocket-casts-ios
# Onboarding: Adds the styling for the updated plus landing view

| 📘 Project: #500 |
|:---:|

This adds the styling for the plus landing view, which includes:
- Image assets for the onboarding flow
- Reusable plus button styles
    - The stroke style can be used as a selectable button that shows a checkmark
- Includes a new common component, a dynamicProperty named ScalingValue (Maybe need a better name)
   - This allows you to define a simple var and have it update dynamically based on the view's frame without needing to do any other calculations.
   - I found this really useful when laying out "absolute" positioned that also needed to appear consistently across various view sizes without needing to do (geometry.size.width * percent) over and over
       - You can see this in use in the `PlusBackgroundGradientView`
        - Also see the Example preview in the ScalingValue 
- This also shows real use of a few components that were added:
    - ScrollViewIfNeeded
    - HorizontalScrollView
Currently the text isn't localized which I'm planning on doing in a future PR, iPad support is also not done either.

## Screenshots
| iPhone SE | iPhone 14 Pro | iPhone 14 Pro Max|
|:---:|:---:|:---:|
|<img width="200" src="https://user-images.githubusercontent.com/793774/200746535-80150b0a-8b8e-4eb2-9e24-1958b613b920.png">|<img width="200" src="https://user-images.githubusercontent.com/793774/200746649-93c2604e-bb2a-4f8c-9cb1-405292c57fb3.png">|<img width="200" src="https://user-images.githubusercontent.com/793774/200747117-4d7a812d-29db-45fc-b4d0-696ab1da564a.png">|

## To test
1. With the feature flag off
2. Go to the Profile tab
3. Tap the Create Folder button
4. Verify the existing behavior works
5. Enable the `onboardingUpdates` FeatureFlag
6. Launch on a larger device
7. Go to the Podcasts view
8. Tap the Folder button
9. ✅ Compare [against the designs](https://www.figma.com/file/PviUP0aiZctOprDuuYRwpb/Onboarding-%26-Upgrade-v2?node-id=406%3A24621) to make sure the view matches and looks good
10. Relaunch on a smaller device like the iPhone SE, and verify the view now scrolls

## Checklist
